File: tests/fixtures/python/d005_hardcoded_keys.py
"""Fixture: should trigger D005 (hardcoded LLM credentials).

Each constructor below passes a literal string as the API key — CWE-798.
Suppresses DF001 / R001 with nemoguardrails + structlog imports so the
golden cleanly shows D005 alone.
"""
import nemoguardrails  # noqa: F401  (suppresses DF001)
import structlog  # noqa: F401  (suppresses R001)

import boto3
import openai
import anthropic
import cohere
from langchain_openai import ChatOpenAI
from langchain_anthropic import ChatAnthropic


def main() -> None:
    client_a = openai.OpenAI(api_key="sk-proj-AAAAAAAAAAAAAAAAAAAAAAAAAAAA")  # D005
    client_b = anthropic.Anthropic(api_key="sk-ant-api03-XXXXXXXXXXXXXXXXX")  # D005
    client_c = cohere.Client("co-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")  # D005
    bedrock = boto3.client(  # D005
        "bedrock-runtime",
        region_name="us-east-1",
        aws_access_key_id="AKIAIOSFODNN7EXAMPLE",
        aws_secret_access_key="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
    )
    chat_a = ChatOpenAI(api_key="sk-proj-BBBBBBBBBBBBBBBBBBBBBBBBBBBB")  # D005
    chat_b = ChatAnthropic(anthropic_api_key="sk-ant-api03-YYYYYYYYYYYYYYY")  # D005

File: tests/fixtures/java/d005_hardcoded_keys.java
// Fixture: should trigger D005-java (hardcoded credentials).
// Each constructor / builder below passes a literal credential string
// — CWE-798. Java port of D005.
package com.example.vuln;

import com.azure.core.credential.AzureKeyCredential;
import dev.langchain4j.model.openai.OpenAiChatModel;
import dev.langchain4j.model.anthropic.AnthropicChatModel;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;

public class HardcodedKeys {

    public OpenAiChatModel openai() {
        return OpenAiChatModel.builder()
                .apiKey("sk-proj-AAAAAAAAAAAAAAAAAAAAAAAAAAAA")  // D005-java
                .modelName("gpt-4o-mini")
                .build();
    }

    public AnthropicChatModel anthropic() {
        return AnthropicChatModel.builder()
                .apiKey("sk-ant-api03-XXXXXXXXXXXXXXXXX")  // D005-java
                .modelName("claude-sonnet-4-5")
                .build();
    }

    public AzureKeyCredential azure() {
        return new AzureKeyCredential("az-key-LITERAL-EXAMPLE-AAAA");  // D005-java
    }

    public AwsBasicCredentials awsBasic() {
        return AwsBasicCredentials.create(  // D005-java
                "AKIAIOSFODNN7EXAMPLE",
                "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
        );
    }

    public StaticCredentialsProvider awsProvider() {
        return StaticCredentialsProvider.create(
                AwsBasicCredentials.create(  // D005-java (inner factory)
                        "AKIAIOSFODNN7EXAMPLE",
                        "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
                )
        );
    }
}
